File: app/src/main/java/com/salah/tracker/data/database/entities/QuranLog.kt
package com.salah.tracker.data.database.entities

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "quran_logs")
data class QuranLog(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val timestamp: Long,
    val surah: String,
    val startAyah: Int,
    val endAyah: Int,
    val startPage: Int,
    val endPage: Int,
    val pagesRead: Int
)

File: flutter_app/lib/data/models/quran_log.dart
class QuranLog {
  final int? id;
  final int timestamp;
  final String surah;
  final int startAyah;
  final int endAyah;
  final int startPage;
  final int endPage;
  final int pagesRead;

  QuranLog({
    this.id,
    required this.timestamp,
    required this.surah,
    required this.startAyah,
    required this.endAyah,
    required this.startPage,
    required this.endPage,
    required this.pagesRead,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'timestamp': timestamp,
      'surah': surah,
      'startAyah': startAyah,
      'endAyah': endAyah,
      'startPage': startPage,
      'endPage': endPage,
      'pagesRead': pagesRead,
    };
  }

  factory QuranLog.fromMap(Map<String, dynamic> map) {
    return QuranLog(
      id: map['id'],
      timestamp: map['timestamp'],
      surah: map['surah'],
      startAyah: map['startAyah'],
      endAyah: map['endAyah'],
      startPage: map['startPage'],
      endPage: map['endPage'],
      pagesRead: map['pagesRead'],
    );
  }
}
